IM_calculation/source_site_dist/src_site_dist.py

```diff
@@ -218,6 +218,64 @@ def calc_rx_ry_GC2(
     return r_x, r_y
 
 
+def calc_rx_ry_GC2_multi_hypocentre(
+    srf_points: np.ndarray,
+    plane_infos: List[Dict],
+    locations: np.ndarray,
+    origin_offsets: np.ndarray = np.asarray([0]),
+):
+    """
+    Vectorised version of the GC2 function along multiple hypocentre locations.
+    Calculates Rx and Ry distances using the cross track and along track distance calculations
+    If there are multiple fault planes the Rx, Ry values are calculated for each fault plane individually, then weighted
+    according to plane length and distance to the location
+    For one fault plane this is the same as the GC1 function
+    :param srf_points: An array with shape (n, 3) giving the lon, lat, depth location of each subfault
+    :param plane_infos: A list of srf header dictionaries, as retrieved from qcore.srf.get_headers with idx=True
+    :param locations: An array with shape (m, 2) giving the lon, lat locations of each location to get Rx, Ry values for
+    :param origin_offsets: An array with shape (o) with the along strike hypocentre locations from the centre of
+    the fault for multiple realisations. If not set then only one origin will be considered at the centre position.
+    :return: Two arrays with shape (o, m) giving the Rx, Ry values for each of the given locations for each hypocentre
+    """
+    # Separate the srf points into the different plane traces
```

Comment:
Would suggest adding some more block comments, since I this is not a trivial function. Whoever has to touch this next will appreciate it. Up to you though.